_posts/2024-08-24-GSoC2024-dingo.md

```diff
@@ -0,0 +1,119 @@
+of the `reduce` function to `True`. These reactions do not affect the value of the objective function when removed.
+
+Reduction with the `PreProcess` class has been tested with various models, some of which are included in dingo’s publication article too. Figures below show the number of remained reactions, after calling the `reduce` function.
+
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/reduction_results_plot.png"><br><br>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/reduction_results_plot2.png"><br><br>
+</center>
+
+
+## Correlated Reactions
+
+- Reactions in biochemical pathways can be positively correlated, negatively correlated, or uncorrelated.
+- Positive correlation: if reaction A is active, then reaction B is also active and vice versa.
+- Negative correlation: if reaction A is active, then reaction B is inactive and vice versa.
+- Zero correlation: The status of reaction A is independent of the status of reaction B and vice versa.
+
+A `correlated_reactions` function that calculates reactions steady states using dingo's `PolytopeSampler` class and creates a correlation matrix was implemented. The correlation matrix is based on the pearson correlation coefficient between pairwise reactions. This function also calculates a copula indicator to filter correlations greater than the pearson cutoff.
+
+A `plot_corr_matrix` function to visualize the correlation matrix as a heatmap plot was implemented too.
+
+This figure illustrates a heatmap from a symmetrical correlation matrix without pearson or indicator filtering:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/corr_matrix_no_cutoffs.png"><br><br>
+</center>
+
+This figure illustrates a heatmap from a triangular correlation matrix with both pearson and indicator filtering:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/corr_matrix_pearson_indicator_triangle.png"><br><br>
+</center>
+
+
+## Clustering
+
+- Clustering based on a dissimilarity matrix reveals groups of reactions with similar correlation values.
+- Reactions within the same cluster may contribute to the same pathways.
+
+A `cluster_corr_reactions` function that hierarchically clusters a correlation matrix 
+alongside a `plot_dendrogram` function that plots a dendrogram were implemented.
+
+This figure illustrates a dendrogram created from a filtered correlation matrix:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/dendrogram_pearson.png"><br><br>
+</center>
+
+Distinct clusters are observed. Graphs will reveal if these clusters interact with other clusters or reactions.
+
+
+## Graphs
+
+- Graphs creation can reveal networks of correlated reactions, potentially corresponding to metabolic pathways.
+
+A `graph_corr_matrix` function that creates graphs from a correlation matrix 
+alongside a `plot_graph` function that plots the graphs were implemented.
```

Comment:
Can you explain how the graph's edges' weights are derived from the correlation matrix?